event-sourcing/event-source-api-integration-test/src/test/java/uk/gov/justice/services/eventsourcing/source/api/EventStreamsFeedIT.java

```diff
@@ -0,0 +1,286 @@
+        final HttpResponse secondPage = feedOf(nextPageUrl);
+
+        assertThat(secondPage.getStatusLine().getStatusCode(), is(200));
+        with(responseBodyOf(secondPage))
+                .assertThat("$.data", hasSize(1))
+                .assertThat("$.data[0].streamId", is(streamId3.toString()))
+                .assertThat("$.data[0].href", containsString(streamId3.toString()));
+
+    }
+
+    @Test
+    public void shouldFollowToThe3rdPage() throws Exception {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+
+
+        eventStreamRepository.insert(new EventStream(streamId1));
+        eventStreamRepository.insert(new EventStream(streamId2));
+        eventStreamRepository.insert(new EventStream(streamId3));
+        eventStreamRepository.insert(new EventStream(streamId4));
+        eventStreamRepository.insert(new EventStream(streamId5));
+
+        eventStreamsFeedService.initialiseWithPageSize(2);
+
+        final HttpResponse firstPage = eventStreamsFeed();
+
+
+        String nextPageUrl = JsonPath.read(responseBodyOf(firstPage), "$.paging.next");
+
+        final HttpResponse secondPage = feedOf(nextPageUrl);
+
+        nextPageUrl = JsonPath.read(responseBodyOf(secondPage), "$.paging.next");
+
+        final HttpResponse thirdPage = feedOf(nextPageUrl);
+
+        assertThat(thirdPage.getStatusLine().getStatusCode(), is(200));
+        final String thirdPageBody = responseBodyOf(thirdPage);
+        with(thirdPageBody)
+                .assertThat("$.data", hasSize(1))
+                .assertThat("$.data[0].streamId", is(streamId5.toString()));
+    }
+
+    @Test
+    public void shouldGoBackToThePreviousPage() throws IOException {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+
+
+        eventStreamRepository.insert(new EventStream(streamId1));
+        eventStreamRepository.insert(new EventStream(streamId2));
+        eventStreamRepository.insert(new EventStream(streamId3));
+
+        eventStreamsFeedService.initialiseWithPageSize(2);
+
+        final HttpResponse firstPage = eventStreamsFeed();
+        String nextPageUrl = JsonPath.read(responseBodyOf(firstPage), "$.paging.next");
+
+        final HttpResponse secondPage = feedOf(nextPageUrl);
+
+        String previousPageUrl = JsonPath.read(responseBodyOf(secondPage), "$.paging.previous");
+        final HttpResponse firstPageAgain = feedOf(previousPageUrl);
+
+        assertThat(firstPageAgain.getStatusLine().getStatusCode(), is(200));
+
+        with(responseBodyOf(firstPageAgain))
+                .assertThat("$.data", hasSize(2))
+                .assertThat("$.data[0].streamId", is(streamId1.toString()))
+                .assertThat("$.data[1].streamId", is(streamId2.toString()));
+
+
+    }
```

Comment:
we've got unit tests for that in EventStreamsFeedServiceTest:
shouldNotReturnLinkToNextPageIfNoMoreRecords
shouldNotReturnLinkToPreviousPageIfOnFirstPage